mappings/net/minecraft/block/Material.mapping

```diff
@@ -2,55 +2,98 @@ CLASS net/minecraft/class_3614 net/minecraft/block/Material
 	FIELD field_15933 PISTON Lnet/minecraft/class_3614;
+		COMMENT Material for piston blocks.
 	FIELD field_15934 SNOW_BLOCK Lnet/minecraft/class_3614;
+		COMMENT Material for full sized snow blocks.
 	FIELD field_15935 PLANT Lnet/minecraft/class_3614;
-	FIELD field_15936 CLAY Lnet/minecraft/class_3614;
+		COMMENT Material for plants such as flowers and crops
+	FIELD field_15936 MOB_DERIVED Lnet/minecraft/class_3614;
+		COMMENT Material for blocks that are a product of mob interactions such as honey or slime, or the infested blocks. Includes clay but not bone blocks.
 	FIELD field_15937 CAKE Lnet/minecraft/class_3614;
+		COMMENT Material for cake.
 	FIELD field_15938 BAMBOO_SAPLING Lnet/minecraft/class_3614;
+		COMMENT Material for bamboo saplings.
 	FIELD field_15939 blocksLight Z
 	FIELD field_15940 blocksMovement Z
-	FIELD field_15941 EARTH Lnet/minecraft/class_3614;
+	FIELD field_15941 TOPSOIL Lnet/minecraft/class_3614;
+		COMMENT Material for the top layer of soil. Path, dirt, podzol, soul soil, farmland and similar.
 	FIELD field_15942 GLASS Lnet/minecraft/class_3614;
+		COMMENT Material for glass and glass-like blocks (includes sea lanterns and conduits).
 	FIELD field_15943 FIRE Lnet/minecraft/class_3614;
+		COMMENT Material for fire blocks.
 	FIELD field_15944 color Lnet/minecraft/class_3620;
-	FIELD field_15945 ORGANIC Lnet/minecraft/class_3614;
+	FIELD field_15945 FODDER Lnet/minecraft/class_3614;
+		COMMENT Food, especially dried hay or feed, for cattle and other livestock. Includes mycelium.
 	FIELD field_15946 BAMBOO Lnet/minecraft/class_3614;
+		COMMENT Material for bamboo.
 	FIELD field_15947 UNDERWATER_PLANT Lnet/minecraft/class_3614;
-	FIELD field_15948 SNOW Lnet/minecraft/class_3614;
-	FIELD field_15949 ANVIL Lnet/minecraft/class_3614;
+		COMMENT Material for underwater plants without the replaceable property.
+	FIELD field_15948 SNOW_LAYER Lnet/minecraft/class_3614;
+		COMMENT Material for non-full blocks of snow. Has the replaceable property.
+	FIELD field_15949 REPAIR_STATION Lnet/minecraft/class_3614;
+		COMMENT Material for blocks that can repair tools, ie. grindstone and anvils.
 	FIELD field_15950 replaceable Z
 	FIELD field_15951 burnable Z
 	FIELD field_15952 BARRIER Lnet/minecraft/class_3614;
+		COMMENT Material for barrier blocks.
 	FIELD field_15953 METAL Lnet/minecraft/class_3614;
-	FIELD field_15954 PUMPKIN Lnet/minecraft/class_3614;
+		COMMENT Material for metal blocks.
+	FIELD field_15954 GOURD Lnet/minecraft/class_3614;
+		COMMENT Material for gourds. Includes the carved pumpkin and jack o' lantern.
 	FIELD field_15955 TNT Lnet/minecraft/class_3614;
+		COMMENT Material for TNT.
 	FIELD field_15956 REPLACEABLE_PLANT Lnet/minecraft/class_3614;
+		COMMENT Material for plant blocks that are replaceable.
 	FIELD field_15957 CARPET Lnet/minecraft/class_3614;
+		COMMENT Material for carpets.
 	FIELD field_15958 ICE Lnet/minecraft/class_3614;
+		COMMENT Material for ice that can melt.
```

Comment:
All ice can melt. Rather specify on the other one that it's for ice that _can't_ melt, or specify here which one to use for use that cannot melt instead.

Reply:
Not all ice in MC can melt. That is included in the JD because it isn't included in the name.

Reply:
All ice can melt (In Real Life) is what I mean. Meltable is the default state and implied by the name "ice".

Saying that this is ice that can melt is redundant. You should rather mention where the behaviour is a divergence from the expected behaviour.

Hence, don't say this is ice that can melt, say the other one is ice that cannot melt and link to it from this one.

Reply:
yes, the name ice implies that it is meltable, but it also implies that it is all ice, which it is not.
A link to the unmeltable kind can be added, but keeping what is already there in the JD will enhance clarity